src/middlewared/middlewared/plugins/docker/state_setup.py

```diff
@@ -49,7 +81,7 @@ async def create_update_docker_datasets(self, docker_ds):
             )
             if not dataset:
                 test_path = os.path.join('/mnt', dataset_name)
```

Comment:
We are not mounting to the app's dataset's natural mountpoint so this logic would not hold - we should account for `IX_APPS_DIR_NAME ` when doing this